### DISCUSSION TITLE
Modifying root exudate flux

### DISCUSSION THREAD

**Discussion by wwieder** [wwieder/biogeochem_testbed_1.1#49] [Ideas]
Current simulations that @eastmanb13 is doing for the Fernow are reducing root exudation with N enrichment. 

Currently root exudates are being calculated as a fraction of fine root NPP https://github.com/wwieder/biogeochem_testbed_1.1/blob/6a10573abe2316517b02ecc2e2e80cac32ff1fde/SOURCE_CODE/casa_cnp.f90#L900

Which means that reducing root exudation actually _increases_ fine root production, which is not what actually occurs in the real world.  In tests reducing belowground allocation with fertilization, the overall CUE of trees increases greatly, as lower N demand from wood production results in much larger than expected changes in total NPP and results in higher litterfall fluxes and high soil respiration, which again is NOT as observed.

The suggestion here would be to have exudates be a function of GPP.
https://github.com/wwieder/biogeochem_testbed_1.1/blob/6a10573abe2316517b02ecc2e2e80cac32ff1fde/SOURCE_CODE/casa_cnp.f90#L167
We need to consider how we're taking this flux out of the model, however. If it comes out before NPP is calculated, then it cannot have an associated N flux.  Therefore, we could have it come out with the calculation of NPP, but with N demand associated with exudate stoichiometry as a parameter that users could modify in the model.  
- If exudates come straight of of GPP before calculating NPP without any N flux, then decreasing exudates with fertilization would increase C available for NPP, but it would also introduce a higher N costs for that NPP (potentially offsetting changes in plant CUE)?  This may also serve to decrease RH fluxes.

A quick way to assess this is suggested below (which will not allow for exudate N fluxes)
'''
        casaflux%fracClabile(np) =min(1.0,max(0.0,(1.0- xNPuptake(np)))) * max(0.0,casaflux%Cnpp(np))/(casaflux%Cgpp(np) +1.0e-10)
        casaflux%Cexudate(np) = max(0.0, casabiome%fracRootExud(veg%iveg(np)) * casaflux%Cgpp(np) )
        casaflux%Cgpp(np)    = casaflux%Cgpp(np) - casaflux%Cexudate(np)
        casaflux%Cnpp(np)    = casaflux%Cnpp(np) - casaflux%fracClabile(np) * casaflux%Cgpp(np)
'''

associated N fluxes are currently being calculated here? I'll comment this out for now, but we can circle back to how to handle 
https://github.com/wwieder/biogeochem_testbed_1.1/blob/6a10573abe2316517b02ecc2e2e80cac32ff1fde/SOURCE_CODE/casa_cnp.f90#L936

**Comment by leoleojkl** [comment]
邮件已收至jkl。祝好。
